app/controllers/application_controller.rb

```diff
@@ -1,2 +1,12 @@
 class ApplicationController < ActionController::Base
+  before_action :set_locale
+
+  private
+  
+  def set_locale
+    I18n.locale = params[:locale] || I18n.default_locale
+  end
+  def default_url_options
+    {locale: I18n.locale}
+  end
```

Comment:
```suggestion
  end
  
  def default_url_options
    {locale: I18n.locale}
  end
```
coding convention : giữa 2 method phân tách bởi 1 line trống